Feedback from Building Botan 3.7.1 in NixOS/nixpkgs

I started to experiment with Botan 3.7.1 in https://github.com/NixOS/nixpkgs/pull/381091, below are my current findings and questions:

- If I build with "--with-tpm2", the tests inside the build jail fail, because no /dev/tpmrm0 or /dev/tmp0 is present. Make these tests optional and only run on device presence?
- When using "--module-policy=bsi", ffi is disabled, but needed for botan-rs or usage as pluging in strongswan IPsec daemon. Is this intentional, to forbid ffi, when in BSI mode?
- With BSI policy and enable module ffi, the ffi_ecdh test fails.

I retested the current master and can confirm, that the botan tests are working in my nix build now with default settings.

If I enable the bsi policy policy together with TPM2 support the following error shows up:

```
 -Wsuggest-override -Wshadow -Wextra-semi -DBOTAN_IS_BEING_BUILT  -I build/include/public -I build/include/internal  -c src/lib/pubkey/classic_mceliece/cmce_keys_internal.cpp -o build/obj/lib/pubkey_classic_mceliece_cmce_keys_internal.o
       > g++ -fPIC -fvisibility=hidden -fstack-protector -m64 -pthread -std=c++20 -D_REENTRANT  -O3 -Wall -Wextra -Wpedantic -Wstrict-aliasing -Wcast-align -Wmissing-declarations -Wpointer-arith -Wcast-qual -Wzero-as-null-pointer-constant -Wnon-virtual-dtor -Wold-style-cast -Wsuggest-override -Wshadow -Wextra-semi -DBOTAN_IS_BEING_BUILT  -I build/include/public -I build/include/internal  -c src/lib/pubkey/classic_mceliece/cmce_matrix.cpp -o build/obj/lib/pubkey_classic_mceliece_cmce_matrix.o
       > src/lib/prov/tpm2/tpm2_key.cpp: In function 'Botan::RSA_PublicKey Botan::TPM2::rsa_pubkey_from_tss2_public(const TPM2B_PUBLIC*)':
       > src/lib/prov/tpm2/tpm2_key.cpp:34:22: error: 'rsa_pubkey_components_from_tss2_public' was not declared in this scope; did you mean 'rsa_pubkey_from_tss2_public'?
       >    34 |                      rsa_pubkey_components_from_tss2_public(public_area));
       >       |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       >       |                      rsa_pubkey_from_tss2_public
```

--- a/src/build-data/policy/modern.txt
+++ b/src/build-data/policy/modern.txt
@@ -8,8 +8,11 @@ sha2_32
 sha2_64
 blake2
 skein
-keccak
+keccak # this is a pre-standard implementation, TODO(Botan4): remove
+shake
 sha3
+
+shake_xof
 
 gcm
 ocb

